XenAdmin/Controls/Ballooning/HostMemoryControls.cs

```diff
@@ -117,10 +125,33 @@ void vm_PropertyChanged(object sender, PropertyChangedEventArgs e)
                 this.Refresh();
         }
 
+        static bool isMetricUpdating;
+        Timer metricUpdaterTimer;
         void host_PropertyChanged(object sender, PropertyChangedEventArgs e)
         {
             if (e.PropertyName == "memory_overhead")
                 this.Refresh();
+            else if (e.PropertyName == "resident_VMs" && MetricUpdater != null)
+            {
+                isMetricUpdating = true;
+                this.Refresh();
+                if (metricUpdaterTimer == null)
+                {
+                    metricUpdaterTimer = new Timer();
+                    metricUpdaterTimer.Interval = 15000;
+                    metricUpdaterTimer.Tick += MetricUpdaterTimer_Tick;
+                }
+                metricUpdaterTimer.Enabled = true;
+            }
+        }
+
+        private void MetricUpdaterTimer_Tick(object sender, EventArgs e)
+        {
+            if (MetricUpdater == null) return;
+            MetricUpdater.Prod();

```

Comment:
It would probably be safer to disable the timer before we do anything else in this method (i.e. move `metricUpdaterTimer.Enabled = false;` here)